src/core/shape_inference/include/fft_base_shape_inference.hpp

```diff
@@ -2,57 +2,90 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 #pragma once
-#include <openvino/op/util/fft_base.hpp>
 
+#include "dimension_util.hpp"
 #include "fft_common_validation.hpp"
 #include "openvino/core/axis_vector.hpp"
+#include "openvino/core/dimension_tracker.hpp"
+#include "openvino/op/util/fft_base.hpp"
 #include "utils.hpp"
 
 namespace ov {
 namespace op {
+
+namespace fft {
+template <class TRShape, typename std::enable_if<std::is_same<TRShape, PartialShape>::value>::type* = nullptr>
+void apply_dims_from_sizes(const util::FFTBase* op,
+                           TRShape& output_shape,
+                           const std::vector<int64_t>& axes,
+                           const ITensorAccessor& ta) {
+    using namespace ov::util;
+    using DimType = typename TRShape::value_type;
+
+    if (const auto output_bounds = get_input_bounds<TRShape, int64_t>(op, 2, ta)) {
+        constexpr auto minus_one_bound = std::make_pair(dim::inf_bound, dim::inf_bound);
+        auto labels =
+            op->get_input_size() > 2 ? op->get_input_source_output(2).get_tensor().get_value_label() : TensorLabel();
+        const bool propagate_labels = !labels.empty();
+        const auto num_of_axes = axes.size();
+        for (size_t i = 0; i < num_of_axes; ++i) {
+            if ((*output_bounds)[i] != minus_one_bound) {
+                output_shape[(axes)[i]] = DimType((*output_bounds)[i].first, (*output_bounds)[i].second);
+                if (propagate_labels) {
```

Comment:
What if axes are labeled like {1, no_label,2}? The set label will throw exception for `no_label`.

Reply:
Good catch, actually not the labels of the `axes`, but the `sizes`, but thanks, fixed.